wifiphisher/common/phishinghttp.py

```diff
@@ -20,6 +23,48 @@ def get(self):
         self.redirect("http://10.0.0.1:8080/")
 
 
+class ValidateHandler(tornado.web.RequestHandler):
+    """
+    Validate the POST requests from client by the uimethods
+    """
+
+    def initialize(self, em):
+        """
+        :param self: A tornado.web.RequestHandler object
+        :param em: An extension manager object
+        :type self: tornado.web.RequestHandler
+        :type em: ExtensionManager
+        :return: None
+        :rtype: None
+        """
+
+        self.em = em
+
+    def post(self):
+        """
+        :param self: A tornado.web.RequestHandler object
+        :type self: tornado.web.RequestHandler
+        :return: None
+        :rtype: None
+        ..note: overide the post method to do the verification
+        """
+
+        json_obj = json_decode(self.request.body)
+        response_to_send = {}
+        verifications = {func.__name__: func for func in
+                         self.em.get_ui_funcs()}
+        # loop all the required verification methods
+        for key in list(json_obj.keys()):
+            if key in verifications:
+                # fire the corresponding varification method
+                result = getattr(uimethods, key)(json_obj[key])
```

Comment:
I think we need to separate the logic:

* One this isuimethods. An extension may use a uimethod for dynamically adjusting the page (upon render). Example: To make the "firmware-upgrade" scenario more effective we could have the number of connected clients somewhere. The extension could capture the number of connected STAs and then using a uimethod pass this to the template.

* Other is backendmethods. These are all the cases where we need some input from the user, e.g. to verify the credentials or as I said in my other comment, there are other cases too (identifying that a user exists in the system).

For the uimethods, the implementation looks complete. For the backendmethods, we need a Handler as the one you created. The handler should use the EM (Extension Manager) to pick all the `backendmethods` though. So, we will need a new decorator here, "@backendmethod" for this. There's no need to put the "uimethods" here as well.